project/src/hooks/useMap.tsx

```diff
@@ -1,31 +1,38 @@
 import {useEffect, useState, MutableRefObject} from 'react';
 import {Map, TileLayer} from 'leaflet';
-import {City} from '../types/offers';
+import {OpenStreetMapProvider} from 'leaflet-geosearch';
 
-function useMap (mapRef: MutableRefObject<HTMLElement | null>, city: City): Map | null {
+function useMap (mapRef: MutableRefObject<HTMLElement | null>, city: string): Map | null {
   const [map, setMap] = useState<Map | null>(null);
+  const provider = new OpenStreetMapProvider();
 
   useEffect(() => {
-    if (mapRef.current !== null && map === null) {
-      const instance = new Map(mapRef.current, {
-        center: {
-          lat: city.location.latitude,
-          lng: city.location.longitude,
-        },
-        zoom: city.location.zoom,
-      });
+    provider.search({ query: `${city} `}).then((result) => {
```

Comment:
```suggestion
    provider.search({ query: `${city} `})
      .then((result) => {
```

лучше then на следующую строку перенести